.ci/Jenkinsfile

```diff
@@ -1,122 +1,12 @@
-        ARTIFACTORY_CREDENTIALS = credentials('artifactory-credentials')
-        CORDA_ARTIFACTORY_USERNAME = "${env.ARTIFACTORY_CREDENTIALS_USR}"
-        CORDA_ARTIFACTORY_PASSWORD = "${env.ARTIFACTORY_CREDENTIALS_PSW}"
-        SNYK_TOKEN = "r3-gradle-plugins"
-        SNYK_COMMANDS = "--all-sub-projects --configuration-matching='^runtimeClasspath\$' --prune-repeated-subdependencies --target-reference='${env.BRANCH_NAME}' --project-tags=Branch='${env.BRANCH_NAME.replaceAll("[^0-9|a-z|A-Z]+","_")}'"
-    }
-    stages {
-        stage('Build') {
-            steps {
-                authenticateGradleWrapper()
-                sh './gradlew --no-daemon --info --no-build-cache clean build'
-            }
-        }
-        
-        stage('Snyk Security Scan') {
-            when {
-                not {
-                    changeRequest()
-                }
-            }
-            steps {
-                snykSecurityScan("${env.SNYK_TOKEN}", "${env.SNYK_COMMANDS}", true, true)
-            }
-        }
-        stage('Publish to Artifactory') {
-            when {
-                expression { params.DO_PUBLISH }
-                beforeAgent true
-            }
-            steps {
-                rtServer (
-                    id: 'R3-Artifactory',
-                    url: 'https://software.r3.com/artifactory',
-                    credentialsId: 'artifactory-credentials'
-                )
-                rtGradleDeployer (
-                    id: 'deployer',
-                    serverId: 'R3-Artifactory',
-                    repo: env.ARTIFACTORY_REPO,
-                )
-                rtGradleRun (
-                    usesPlugin: true,
-                    useWrapper: true,
-                    switches: "--no-daemon --info -s",
-                    tasks: 'artifactoryPublish',
-                    deployerId: 'deployer',
-                    buildName: env.ARTIFACTORY_BUILD_NAME
-                )
-                rtPublishBuildInfo (
-                    serverId: 'R3-Artifactory',
-                    buildName: env.ARTIFACTORY_BUILD_NAME
-                )
-            }
-        }
-    }
-    post {
-        always {
-            script {
-                if(!env.CHANGE_ID) {
-                    snykSecurityScan.generateHtmlElements()
-                }
-            }
-            junit testResults: '**/build/test-results/**/*.xml', keepLongStdio: true, allowEmptyResults: true
-        }
-        cleanup {
-            deleteDir()
-        }
-    }
-}
-
-def isReleaseBranch() {
-    return (env.BRANCH_NAME =~ /^release\/.*$/)
-}
-
-def isMainBranch() {
-    return (env.BRANCH_NAME =~ /^master$/)
-}
+@Library('corda-shared-build-pipeline-steps@5.0') _
+
+cordaPipeline(
+    dedicatedJobForSnykDelta: false,
+    javaVersion: '8',
+    publishToMavenS3Repository: true,
+    publishRepoPrefix: 'corda',
+    slimBuild: true,
+    stableUnstableRepoPattern: false,
+    nexusAppId :'corda-gradle-plugins-7.0.0',
```

Comment:
This isn't a good name. What happens when we upgrade the plugins to version 7.0.x, 7.1.x or even 8.x?

Will this pipeline be used for the `release/5.0`, `release/5.1` and release/6.0` branches too?

Reply:
Each time there is a version change, a new "application id" needs to be created on the Nexus server side, otherwise the previous versions scans will be overwritten.

These pipeline changes are mainly to facilitate a smooth C5 release so there are no current plans to backport to release/5 etc.

Also worth nothing the Business plan is to retire Nexus fully with in the next year in favor of Snyk / other tooling.